subprojects/plugin-use/src/main/java/org/gradle/plugin/management/internal/DefaultPluginResolveDetails.java

```diff
@@ -0,0 +1,26 @@
+package org.gradle.plugin.management.internal;
+
+import org.gradle.api.Action;
+import org.gradle.plugin.management.ConfigurablePluginRequest;
+import org.gradle.plugin.management.PluginRequest;
+import org.gradle.plugin.management.PluginResolveDetails;
+
+public class DefaultPluginResolveDetails implements PluginResolveDetails {
+
+    private final ConfigurablePluginRequest pluginRequest;
```

Comment:
This needs to have two fields. The `requested`, immutable plugin request and the `target`, also immutable plugin request.